File: Striver Important Interview Questions/Day 1 - Arrays/best-time-to-buy-and-sell-stock.cpp

// https://leetcode.com/problems/best-time-to-buy-and-sell-stock/description/

class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int currMin = INT_MAX;
        int profit = INT_MIN;

        for(auto price : prices){
            currMin = min(currMin, price);
            profit = max(profit, price-currMin);
        }

        return profit;   
    }
};

File: Neetcode-150-js/Sliding Window/best-time-to-buy-and-sell-stock.js

// https://leetcode.com/problems/best-time-to-buy-and-sell-stock/description/

/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let currMin = Number.MAX_SAFE_INTEGER;
    let profit = Number.MIN_SAFE_INTEGER;

    for(let price of prices){
        currMin = Math.min(currMin, price);
        profit = Math.max(profit, price - currMin);
    }

    return profit;
};